x-pack/plugins/ml/public/application/components/ml_inference/components/add_inference_pipeline_footer.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { AddInferencePipelineSteps } from '../types';
+import { BACK_BUTTON_LABEL, CANCEL_BUTTON_LABEL, CONTINUE_BUTTON_LABEL } from '../constants';
+
+interface Props {
+  isConfigureStepValid: boolean;
+  isPipelineDataValid: boolean;
+  pipelineCreated: boolean;
+  creatingPipeline: boolean;
+  step: AddInferencePipelineSteps;
+  onClose: () => void;
+  onCreate: () => void;
+  setStep: React.Dispatch<React.SetStateAction<AddInferencePipelineSteps>>;
+}
+
+export const AddInferencePipelineFooter: FC<Props> = ({
+  isConfigureStepValid,
+  isPipelineDataValid,
+  creatingPipeline,
+  pipelineCreated,
+  onClose,
+  onCreate,
+  step,
+  setStep,
+}) => {
+  let nextStep: AddInferencePipelineSteps | undefined;
```

Comment:
You could break out the code to set these `lets` in a function, will also allow to write unit tests for it more easily.

Reply:
Updated in 4b1791c5523f98483b24f0ca0dd3f1e88b855a50